src/Properties/ModelPropertyExtension.php

```diff
@@ -44,270 +44,147 @@ public function hasProperty(ClassReflection $classReflection, string $propertyNa
             return false;
         }
 
-        if (count($this->tables) === 0) {
-            // First try to create tables from squashed migrations, if there are any
-            // Then scan the normal migration files for further changes to tables.
-            $tables = $this->squashedMigrationHelper->initializeTables();
-
-            $this->tables = $this->migrationHelper->initializeTables($tables);
-        }
-
-        if ($propertyName === 'id') {
-            return true;
+        if (! $this->migrationsLoaded()) {
+            $this->loadMigrations();
         }
 
-        $modelName = $classReflection->getNativeReflection()->getName();
-
         try {
-            $reflect = $this->reflectionProvider->getClass($modelName);
-
             /** @var Model $modelInstance */
-            $modelInstance = $reflect->getNativeReflection()->newInstanceWithoutConstructor();
-
-            $tableName = $modelInstance->getTable();
-        } catch (ClassNotFoundException|\ReflectionException $e) {
+            $modelInstance = $classReflection->getNativeReflection()->newInstanceWithoutConstructor();
+        } catch (\ReflectionException $e) {
             return false;
         }
 
+        if ($propertyName === $modelInstance->getKeyName()) {
+            return true;
+        }
+
+        $tableName = $modelInstance->getTable();
+
         if (! array_key_exists($tableName, $this->tables)) {
             return false;
         }
 
-        if (! array_key_exists($propertyName, $this->tables[$tableName]->columns)) {
-            return false;
+        return array_key_exists($propertyName, $this->tables[$tableName]->columns);
+    }
+
+    private function hasAccessor(ClassReflection $classReflection, string $propertyName): bool
+    {
+        $propertyNameStudlyCase = Str::studly($propertyName);
+
+        if ($classReflection->hasNativeMethod("get{$propertyNameStudlyCase}Attribute")) {
+            return true;
         }
 
-        $this->castPropertiesType($modelInstance);
+        $propertyNameCamelCase = Str::camel($propertyName);
 
-        $column = $this->tables[$tableName]->columns[$propertyName];
+        if ($classReflection->hasNativeMethod($propertyNameCamelCase)) {
+            $methodReflection = $classReflection->getNativeMethod($propertyNameCamelCase);
 
-        [$readableType, $writableType] = $this->getReadableAndWritableTypes($column, $modelInstance);
+            if ($methodReflection->isPublic() || $methodReflection->isPrivate()) {
+                return false;
+            }
 
-        $column->readableType = $readableType;
-        $column->writeableType = $writableType;
+            $returnType = ParametersAcceptorSelector::selectSingle($methodReflection->getVariants())->getReturnType();
+
+            if (! (new ObjectType(Attribute::class))->isSuperTypeOf($returnType)->yes()) {
+                return false;
+            }
+
+            return true;
```

Comment:
I think we can simplify it to `return (new ObjectType(Attribute::class))->isSuperTypeOf($returnType)->yes()`